Makefile

```diff
@@ -194,49 +200,50 @@ $(BINDIR)/validate-trust-package: cmd/validate-trust-package/main.go pkg/fspkg/p
 	cd hack/tools && go build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
 
-$(BINDIR)/boilersuite: | $(BINDIR)
+$(BINDIR)/boilersuite-$(BOILERSUITE_VERSION)/boilersuite: | $(BINDIR)/boilersuite-$(BOILERSUITE_VERSION)
 	cd hack/tools && go build -o $@ github.com/cert-manager/boilersuite
 
-$(BINDIR)/kind: | $(BINDIR)
+$(BINDIR)/kind-$(KIND_VERSION)/kind: | $(BINDIR)/kind-$(KIND_VERSION)
 	cd hack/tools && go build -o $@ sigs.k8s.io/kind
 
-$(BINDIR)/helm-docs: | $(BINDIR)
+$(BINDIR)/helm-docs-$(HELM_DOCS_VERSION)/helm-docs: | $(BINDIR)/helm-docs-$(HELM_DOCS_VERSION)
 	cd hack/tools && go build -o $@ github.com/norwoodj/helm-docs/cmd/helm-docs
 
-$(BINDIR)/helm: $(BINDIR)/helm-v$(HELM_VERSION)-$(OS)-$(ARCH).tar.gz | $(BINDIR)
-	tar xfO $< $(OS)-$(ARCH)/helm > $@
-	chmod +x $@
+$(BINDIR)/helm-$(HELM_VERSION)/helm: $(BINDIR)/helm-$(HELM_VERSION)/helm-v$(HELM_VERSION)-$(OS)-$(ARCH).tar.gz | $(BINDIR)
+	tar xfO $< $(OS)-$(ARCH)/helm > $@ && chmod +x $@
 
-$(BINDIR)/helm-v$(HELM_VERSION)-$(OS)-$(ARCH).tar.gz: | $(BINDIR)
+$(BINDIR)/helm-$(HELM_VERSION)/helm-v$(HELM_VERSION)-$(OS)-$(ARCH).tar.gz: | $(BINDIR)/helm-$(HELM_VERSION)
 	curl -o $@ -LO "https://get.helm.sh/helm-v$(HELM_VERSION)-$(OS)-$(ARCH).tar.gz"
 
-$(BINDIR)/ginkgo: $(BINDIR)/ginkgo-$(GINKGO_VERSION)/ginkgo | $(BINDIR)
-	cp -f $< $@
-
 $(BINDIR)/ginkgo-$(GINKGO_VERSION)/ginkgo: | $(BINDIR)
 	GOBIN=$(dir $@) go install github.com/onsi/ginkgo/v2/ginkgo@$(GINKGO_VERSION)
 
-$(BINDIR)/kubectl: | $(BINDIR)
-	curl -o $@ -LO "https://storage.googleapis.com/kubernetes-release/release/$(shell curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/$(OS)/$(ARCH)/kubectl"
-	chmod +x $@
+$(BINDIR)/kubectl-$(KUBECTL_VERSION)/kubectl: | $(BINDIR)/kubectl-$(KUBECTL_VERSION)
+	curl -o $@ -L "https://storage.googleapis.com/kubernetes-release/release/v$(KUBECTL_VERSION)/bin/$(OS)/$(ARCH)/kubectl" && chmod +x $@
+
+$(BINDIR)/kubebuilder-$(KUBEBUILDER_TOOLS_VERISON)/etcd: $(BINDIR)/kubebuilder-$(KUBEBUILDER_TOOLS_VERISON)/envtest-bins.tar.gz | $(BINDIR)/kubebuilder-$(KUBEBUILDER_TOOLS_VERISON)
+	tar xfO $< kubebuilder/bin/etcd > $@ && chmod +x $@
+
+$(BINDIR)/kubebuilder-$(KUBEBUILDER_TOOLS_VERISON)/kube-apiserver: $(BINDIR)/kubebuilder-$(KUBEBUILDER_TOOLS_VERISON)/envtest-bins.tar.gz | $(BINDIR)/kubebuilder-$(KUBEBUILDER_TOOLS_VERISON)
+	tar xfO $< kubebuilder/bin/kube-apiserver > $@ && chmod +x $@
 
-$(BINDIR)/kubebuilder/bin/kube-apiserver: | $(BINDIR)/kubebuilder
-	curl -sSLo $(BINDIR)/envtest-bins.tar.gz "https://storage.googleapis.com/kubebuilder-tools/kubebuilder-tools-$(KUBEBUILDER_TOOLS_VERISON)-$(OS)-$(ARCH).tar.gz"
-	tar -C $(BINDIR)/kubebuilder --strip-components=1 -zvxf $(BINDIR)/envtest-bins.tar.gz
+$(BINDIR)/kubebuilder-$(KUBEBUILDER_TOOLS_VERISON)/envtest-bins.tar.gz: | $(BINDIR)/kubebuilder-$(KUBEBUILDER_TOOLS_VERISON)
+	curl -sSL -o $@ "https://storage.googleapis.com/kubebuilder-tools/kubebuilder-tools-$(KUBEBUILDER_TOOLS_VERISON)-$(OS)-$(ARCH).tar.gz"
```

Comment:
I think this might be possible to simplify by introducing https://pkg.go.dev/sigs.k8s.io/controller-runtime/tools/setup-envtest, but we could create a follow-up PR for that.

Reply:
I wasn't aware of that tool! Could be a good addition for sure - could save a bit of downloading in dev environments especially across our various projects!